### PR TITLE
[AIRFLOW-1575] Add AWS Kinesis Firehose Hook for inserting batch records

### PR DESCRIPTION
**Pull request by sid88in**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1575

### Description
- [x] Here are some details about my PR, (no UI changes)
Description is in above Jira. This PR creates a hook in contrib directory to integrate airflow with aws kinesis firehose for inserting batch items. 

### Tests
- [x] My PR adds the following unit tests 

1. test_aws_firehose_hook
This unit test checks airflow integration with AWS Kinesis Firehose. 1. checks if the connection is established using client API 2. checks batch insertion of data. We have to first create a kinesis firehose stream and then insert data

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by sid88in** [comment]
@bolkedebruin resolved commits. thanks

**Comment by sid88in** [comment]
@artwr @criccomini (+ @saguziel @bolkedebruin) seems like you guys reviewed similar AWS related PR - https://github.com/apache/incubator-airflow/pull/1630

Would be great if you could help review my PR. Any inputs are welcome (this is my first contribution to airflow) :)

**Comment by bolkedebruin** [comment]
Can you please stop opening/closing your PRs? If you want to trigger a build please do this on your own Github account. Now you are taking build time away from other PRs and spamming email boxes.

**Comment by sid88in** [comment]
@bolkedebruin sorry, I'll take care of this. I didn't know right way to trigger builds (new to the process).

**Comment by sid88in** [comment]
@bolkedebruin @saguziel is this PR good to go?

File: airflow/contrib/hooks/__init__.py [hunk 1]
```diff
@@ -47,7 +47,8 @@
     'cloudant_hook': ['CloudantHook'],
     'fs_hook': ['FSHook'],
     'wasb_hook': ['WasbHook'],
-    'gcp_pubsub_hook': ['PubSubHook']
+    'gcp_pubsub_hook': ['PubSubHook'],
+    'aws_dynamodb_hook': ['AwsDynamoDBHook']
```
Comment (r39132) [inline airflow/contrib/hooks/__init__.py:51]: `aws_dynamodb_hook`? The hook in this PR refers to the AWS firehose hook correct?

**Review by r39132 — CHANGES_REQUESTED** [review]
Please see comments.. rebase and I'll test this out.

**Comment by sid88in** [comment]
@r39132 yeah this requires rebasing. Maybe I'll create a fresh PR. Will look into it this week.

Thanks for revisiting this PR.

**Comment by r39132** [comment]
Let's keep this PR open.. just rebase and ensure that you register the appropriate hook in the file above.

**Comment by r39132** [comment]
@sid88in Ok.. feel free to open a fresh PR. Closing this.

**Comment by sid88in** [comment]
thanks @r39132 - I have created new PR here - https://github.com/apache/incubator-airflow/pull/3275

**Pull request closed without merging**